src/plots/plots.js

```diff
@@ -527,6 +563,9 @@ plots.supplyDataDefaults = function(dataIn, dataOut, layout) {
         basePlotModules = layout._basePlotModules = [],
         cnt = 0;
 
+    // Make this idempotent by emptying the existing array:
+    dataOut.length = 0;
```

Comment:
Array.length is [listed as writable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length). Works in IE11:

<img width="257" alt="2016-08-24 10 44 43" src="https://cloud.githubusercontent.com/assets/572717/17934983/d461a8d8-69e7-11e6-8e02-de7a09c02032.png">